### PR TITLE
Migrate to null safety

### PR DESCRIPTION
**Pull request by GaspardMerten**
Small code changes to make this package compatible with the new null-safety mechanism.

### CODE REVIEW

**Comment by GaspardMerten** [comment]
@nex3 could it to publish this version of the package to pub.dev? It would allow me to work on the migration of other packages that depend on this one.

**Comment by nex3** [comment]
I'm no longer on the Dart team, I suggest asking another maintainer.

**Comment by GaspardMerten** [comment]
I will try my luck with @athomas  then :D ?

File: lib/test_process.dart [hunk 1]
```diff
@@ -30,13 +30,13 @@ class TestProcess {
   ///
   /// A copy of the underlying stream can be retreived using [stdoutStream].
   StreamQueue<String> get stdout => _stdout;
-  StreamQueue<String> _stdout;
+  late StreamQueue<String> _stdout;
```
Comment (natebosch) [inline lib/test_process.dart]: I think we can use `late final` here to simplify.

```dart
  late final StreamQueue<String> stdout = StreamQueue(stdoutStream());
```

This should let us drop the private `_stdout` and remove line 127. The same can happen with `stderr`.

Comment (GaspardMerten) [reply lib/test_process.dart]: Yep corrected! (for both stderr & stdout), I also fixed some typos

**Comment by kevmoo** [comment]
@GaspardMerten – need to remove the legacy SDK test bit from the workflow.

File: lib/test_process.dart [hunk 2]
```diff
@@ -28,15 +28,13 @@ class TestProcess {
 
   /// A [StreamQueue] that emits each line of stdout from the process.
   ///
-  /// A copy of the underlying stream can be retreived using [stdoutStream].
-  StreamQueue<String> get stdout => _stdout;
-  StreamQueue<String> _stdout;
+  /// A copy of the underlying stream can be retrieved using [stdoutStream].
+  late final StreamQueue<String> stdout = StreamQueue(stdoutStream());
```
Comment (kevmoo) [inline lib/test_process.dart:32]: @natebosch – should we care about the whole `late final` creates a setter thingy here? Keep the properties?

Comment (kevmoo) [reply lib/test_process.dart:32]: `late final` here is great! nevermind.

https://github.com/dart-lang/language/blob/master/accepted/future-releases/nnbd/feature-specification.md#late-fields-and-variables

**Comment by GaspardMerten** [comment]
> @GaspardMerten – need to remove the legacy SDK test bit from the workflow.

Done!

File: pubspec.yaml [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
 environment:
-  sdk: '>=2.1.0 <3.0.0'
+  sdk: ">=2.12.0-0 <3.0.0"
 
 dependencies:
-  async: ^2.0.0
-  meta: ^1.0.0
-  path: ^1.0.0
-  test: ^1.0.0
+  async: ^2.5.0-nullsafety.0
```
Comment (kevmoo) [inline pubspec.yaml]: many of these are stable now. Drop the `-nullsafety` bits where you can!

File: pubspec.yaml [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
+  meta: ^1.3.0-nullsafety.0
+  path:  ^1.8.0-nullsafety.0
+  test: ^1.16.0-nullsafety.0
 
 dev_dependencies:
-  pedantic: ^1.0.0
-  test_descriptor: ^1.0.0
+  pedantic: ^1.10.0-nullsafety.0
+  test_descriptor: ^2.0.0-nullsafety
```
Comment (kevmoo) [inline pubspec.yaml]: Add a trailing newline here!

File: lib/test_process.dart [hunk 5]
```diff
@@ -112,7 +110,7 @@ class TestProcess {
   /// This is protected, which means it should only be called by subclasses.
   @protected
   TestProcess(Process process, this.description,
-      {Encoding encoding, bool forwardStdio = false})
+      {required Encoding encoding, bool forwardStdio = false})
```
Comment (kevmoo) [inline lib/test_process.dart]: Are you sure this is required here? Or should it just default to `utf8` like above?

File: .github/workflows/test-package.yml [hunk 6]
```diff
@@ -58,28 +58,4 @@ jobs:
...
-      - uses: dart-lang/setup-dart@v0.3
-        with:
-          sdk: ${{ matrix.sdk }}
-      - id: install
-        name: Install dependencies
-        run: pub get
-      - name: Run VM tests
-        run: pub run test --platform vm
-        if: always() && steps.install.outcome == 'success'
+        if: always() && steps.install.outcome == 'success'
```
Comment (kevmoo) [inline .github/workflows/test-package.yml]: trailing newline!

Comment (GaspardMerten) [reply .github/workflows/test-package.yml]: Fixed.

File: lib/test_process.dart (hunk 5 above)
Comment (GaspardMerten) [reply lib/test_process.dart]: Indeed it would be more practical! It is now defaulting to utf8 as you proposed.

File: pubspec.yaml (hunk 3 above)
Comment (GaspardMerten) [reply pubspec.yaml]: Yep, I updated all of them.

File: pubspec.yaml [hunk 7]
```diff
@@ -1,18 +1,18 @@
 name: test_process
-version: 1.0.6-dev
+version: 2.0.0-nullsafety.0
```
Comment (kevmoo) [inline pubspec.yaml]: @natebosch – might as well just go straight to stable, right?

Comment (natebosch) [reply pubspec.yaml]: Yeah we may as well.

```suggestion
version: 2.0.0
```

Still needs to be validated with google3, but we can have it prepped for a stable publish.

**Review by natebosch — APPROVED** [review]

**Pull request merged**